Core/GDCore/IDE/ObjectAssetSerializer.cpp

```diff
@@ -0,0 +1,216 @@
+  objectAssetsElement.ConsiderAsArrayOf("objectAsset");
+  SerializerElement &objectAssetElement =
+      objectAssetsElement.AddChild("objectAsset");
+
+  cleanObject->SerializeTo(objectAssetElement.AddChild("object"));
+
+  SerializerElement &resourcesElement =
+      objectAssetElement.AddChild("resources");
+  resourcesElement.ConsiderAsArrayOf("resource");
+  auto &resourcesManager = project.GetResourcesManager();
+  gd::ResourcesInUseHelper resourcesInUse(resourcesManager);
+  cleanObject->GetConfiguration().ExposeResources(resourcesInUse);
+  for (auto &&newResourceName : resourcesInUse.GetAllResources()) {
+    if (newResourceName.length() == 0) {
+      continue;
+    }
+    auto &resource = resourcesManager.GetResource(
+        resourcesNameReverseMap.find(newResourceName) !=
+                resourcesNameReverseMap.end()
+            ? resourcesNameReverseMap[newResourceName]
+            : newResourceName);
+    SerializerElement &resourceElement = resourcesElement.AddChild("resource");
+    resource.SerializeTo(resourceElement);
+    // Override name and file because the project and the asset don't use the
+    // same one.
+    resourceElement.SetAttribute("kind", resource.GetKind());
+    resourceElement.SetAttribute("name", newResourceName);
+    auto &oldFilePath = resource.GetFile();
+    resourceElement.SetAttribute("file",
+                                 resourcesFileNameMap.find(oldFilePath) !=
+                                         resourcesFileNameMap.end()
+                                     ? resourcesFileNameMap[oldFilePath]
+                                     : oldFilePath);
+  }
+
+  SerializerElement &requiredExtensionsElement =
+      objectAssetElement.AddChild("requiredExtensions");
+  requiredExtensionsElement.ConsiderAsArrayOf("requiredExtension");
+  if (project.HasEventsFunctionsExtensionNamed(extensionName)) {
+    SerializerElement &requiredExtensionElement =
+        requiredExtensionsElement.AddChild("requiredExtension");
+    requiredExtensionElement.SetAttribute("extensionName", extensionName);
+    requiredExtensionElement.SetAttribute("extensionVersion", "1.0.0");
+  }
+
+  // TODO This can be removed when the asset script no longer require it.
+  SerializerElement &customizationElement =
+      objectAssetElement.AddChild("customization");
+  customizationElement.ConsiderAsArrayOf("empty");
+}
+
+void ObjectAssetSerializer::RenameObjectResourceFiles(
+    gd::Project &project, gd::Object &object,
+    const gd::String &destinationDirectory, const gd::String &objectFullName,
+    std::map<gd::String, gd::String> &resourcesFileNameMap,
+    std::map<gd::String, gd::String> &resourcesNameReverseMap) {
+  gd::AssetResourcePathCleaner assetResourcePathCleaner(
+      project.GetResourcesManager(), resourcesFileNameMap,
+      resourcesNameReverseMap);
+  object.GetConfiguration().ExposeResources(assetResourcePathCleaner);
+
+  if (object.GetConfiguration().GetType() == "Sprite") {
```

Comment:
Add a comment explaining why you do all this work specifically for Sprite. It's just an "optimization" but it would work without this, right?